### PR TITLE
Add a Carousel Comp & Improve BadgePreview

### PR DESCRIPTION
**Pull request by FedeMadoery**
WIP

- [x] Add Carousel stories
- [x] Add Carousel comp
- [x] Clean BadgePreview
- [x] Add fixed sized to the BadgePreview
- [x] Add "glare" effect on BadgePreview
- [x] Add Steper comp
- [x] Add Steper stories
- [x] Add `.d.ts` on each main `.scss` file

Carousel Preview Example:

Its automatically moving to the next element every 3secs, just for testing. But that is how it looks like

(this video is not up to date, consider it as an example) 

https://user-images.githubusercontent.com/21246763/201956626-5e9507e5-4897-4d4b-a576-5c620e5d6723.mov

Steper Preview Example:

https://user-images.githubusercontent.com/21246763/203072336-bacfcecb-d808-4b43-818c-70b3cc0b482f.mov

Badge Sizes Preview:

https://user-images.githubusercontent.com/21246763/203072879-82c7642f-37a6-4e74-8d09-7667c84a07c9.mov

### CODE REVIEW

**Comment by ramabit** [comment]
@FedeMadoery nice features, I have 2 things to comment:
1) what is the advantage of generating all those types for the scss variables?
2) the component 'Steper' is a typo, it should be named 'Stepper'
I will test everything locally and let you know if I find any trouble

File: src/components/molecules/Carousel/carousel.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  transform: translateZ(-288px);
+  transform-style: preserve-3d;
+  transition: transform 1s;
+}
+
+.carousel__element {
+  position: absolute;
+  left: 0px;
+  top: 0px;
+  // Si saco este borde, tengo un bug con el border de la badge -- raaaro
```
Comment (Agupane) [inline src/components/molecules/Carousel/carousel.scss:23]: 👀👀

File: src/components/molecules/Steper/Steper.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  color?: Colors
+}
+
+const defaultSteperProps = {
+  title: null,
+  elements: [],
+  minHeight: 0,
+}
+
+export const Steper = ({ elements, title, minHeight, color }: SteperProps = defaultSteperProps) => {
```
Comment (Agupane) [inline src/components/molecules/Steper/Steper.tsx:21]: Stepper has a typo, should be Stepper not Steper

File: src/components/molecules/Steper/Steper.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      </Box>
+    </Box>
+  )
+}
+
+type SteperTitleProps = {
+  color?: Colors
+}
+
+const SteperTitle = ({ children, color }: React.PropsWithChildren<SteperTitleProps>) => {
```
Comment (Agupane) [inline src/components/molecules/Steper/Steper.tsx:71]: Same

File: src/components/molecules/Carousel/Carousel.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+export const Carousel = (props: CarouselProps = defaultCarouselProps) => {
+  const [selectedCell, setSelectedCell] = useState(0)
+  const [intervalId, setIntervalId] = useState<NodeJS.Timer | null>(null)
+
+  const carouselRef: RefObject<HTMLDivElement> = createRef<HTMLDivElement>()
+  // Refs to move the elemenets
+  const elementRefs: RefObject<HTMLDivElement>[] = props.elements.map(() => createRef<HTMLDivElement>())
+
+  function getHelperData() {
```
Comment (Agupane) [inline src/components/molecules/Carousel/Carousel.tsx:27]: Can this logic moved to a hook? or makes no sense?

File: src/components/molecules/Carousel/Carousel.tsx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    const cellSize = props.elementWidth + 16
+
+    const radius = Math.round(cellSize / 2 / Math.tan(Math.PI / cellCount))
+
+    return { cellCount, theta, radius, cellSize }
+  }
+
+  function rotateCarousel() {
+    const { theta, radius } = getHelperData()
+    const angle = theta * selectedCell * -1
```
Comment (Agupane) [inline src/components/molecules/Carousel/Carousel.tsx:39]: I know it's complicate but would be possible to add a comment to explain the logic here? in two weeks we will forgot about this

File: src/components/molecules/Carousel/Carousel.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    const carousel = carouselRef.current
+    if (!carousel) return
+    carousel.style.transform = `translateZ(${-radius}px) rotateY(${angle}deg)`
+  }
+
+  function changeCarousel() {
```
Comment (Agupane) [inline src/components/molecules/Carousel/Carousel.tsx:45]: Same

File: src/components/molecules/Carousel/carousel.scss (hunk 1 above)
Comment (FedeMadoery) [reply src/components/molecules/Carousel/carousel.scss:23]: Yep, It was just a reminder for me, I need to remove it... Good catch

**Comment by FedeMadoery** [comment]
> @FedeMadoery nice features, I have 2 things to comment:
> 
> 1. what is the advantage of generating all those types for the scss variables?
> 2. the component 'Steper' is a typo, it should be named 'Stepper'
> 
> I will test everything locally and let you know if I find any trouble

With all of that types, you can create literals for each set of variables, like Colors. So you will be able to restrict the colors that your component would receive as props, with just an string, also this allow us to just recive colors from our own pallete as an string. You can see a very simple example inside the Stepper if I'm not explaining it well.

File: src/components/molecules/Carousel/Carousel.tsx (hunk 5 above)
Comment (FedeMadoery) [reply src/components/molecules/Carousel/Carousel.tsx:39]: I can give a try, its all mathematical stuff

File: src/components/molecules/Carousel/Carousel.tsx (hunk 4 above)
Comment (FedeMadoery) [reply src/components/molecules/Carousel/Carousel.tsx:27]: It's doable, but I think that doesn't make sense, it is tied to the logic of the carousel, so you wont use it again on another comp.

**Comment by FedeMadoery** [comment]
I did my best explaining the carousel logic, let me know if its not clear and we can have a talk about it.

If you read the comments with this img in mind, everything will be more clear 

image

**Review by Agupane — APPROVED** [review]
Looks great and the explanation was clear I will approve but lets wait for Fer tests, btw the new components looks very good, good job 👍

**Pull request merged**